src/__internal__/focus-trap/focus-trap.spec.js

```diff
@@ -34,32 +37,78 @@ describe("FocusTrap", () => {
   });
   const otherKey = new KeyboardEvent("keydown", { keyCode: 32 });
 
-  describe("when autoFocus is false", () => {
-    let wrapper;
+  describe("triggerRefocusFlag", () => {
+    afterEach(() => {
+      wrapper.unmount();
+    });
+
+    it("refocuses the last element that had focus within the trap when flag is set", () => {
+      const focusMock = jest.fn();
 
-    beforeEach(() => {
       wrapper = mount(
-        <MockComponent autoFocus={false}>
+        <MockComponent autoFocus={false} triggerRefocusFlag={false}>
           <button type="button">Test button One</button>
-          <input type="text" />
+          <input onFocus={focusMock} type="text" />
         </MockComponent>,
         { attachTo: htmlElement }
       );
+      act(() => {
+        document.querySelectorAll("input")[0].focus();
+      });
+      act(() => {
+        document.querySelectorAll("input")[0].blur();
+      });
+      act(() => {
+        wrapper.setProps({ triggerRefocusFlag: true });
+      });
+      wrapper.update();
+      expect(focusMock).toHaveBeenCalledTimes(2);
     });
 
-    afterEach(() => {
-      wrapper.unmount();
+    it("refocuses the first element within the trap when flag is set", () => {
```

Comment:
This is just the same test as the first one, with focus on a different element - I think we just need to set the flag here without manually triggering any focus and checking that the first focusable element received focus

Reply:
Yeah good point there's no need to focus the element like in the test above, I'll remove it and update it to say the callback is called once